### DISCUSSION TITLE
When i view password it shows the scrambled text in macos silicon but other keepass software works fine

### DISCUSSION THREAD

**Discussion by khanakia** [keepassxreboot/keepassxc#10619] [Q&A]
When i view the password field it shows this on macos silicon

However I tried other keepass view software that shows the password correctly
```
ί���K�
```
![Screenshot 2024-04-24 at 7 24 55 AM](https://github.com/keepassxreboot/keepassxc/assets/2174170/2654d60c-48fc-4bc1-887d-03b777faba25)

**Comment by droidmonkey** [comment]
Huh?

**Reply by khanakia** [reply]
Check if i check the password with keyweb.app it works fine but when I open the file with the keepassxc and view password it shows scrambled text
![Screenshot 2024-04-24 at 9 08 07 AM](https://github.com/keepassxreboot/keepassxc/assets/2174170/aeebb7a2-13e2-4023-85e6-187c50ea0dc2)


You can download the file here https://drive.google.com/file/d/19D5Hggx2jfTVEHo2t4mQghgIESXiyxXx/view?usp=sharing and test yourself
pass is `admin`

**Reply by droidmonkey** [reply]
None of these screenshots is KeePassXC

**Reply by khanakia** [reply]
The first screenshot is from KeePassXC but it does not lets you take screenshots seems some kind of protection.

The other screenshot I sent from the other app `keyweb` app I clearly mentioned to show that view password works there

**Reply by droidmonkey** [reply]
View -> Allow Screencapture

**Reply by khanakia** [reply]
Check this video please

https://github.com/keepassxreboot/keepassxc/assets/2174170/11747b4d-ceb4-4f01-8486-e6433ab2a90b

**Comment by droidmonkey** [comment]
Did you create that password within keepassxc or keeweb?

**Reply by khanakia** [reply]
I am using Jetbrains Datagrip software it creates the `datagrip2023_3.kdbx`

`keeweb` i used to open the file in view mode so I can verify that i can view the password but not in keepassxc.

**Reply by droidmonkey** [reply]
this is a problem with Datagrip, not keepassxc. datagrip is clearly not writing a correct kdbx file, it is not using the inner stream encryption. Keyweb is also not honoring that protection either.

**Reply by khanakia** [reply]
I just tried the official windows software from https://keepass.info/download.html and tested it in Windows and there it also works fine.

Then how come only `keepassxc` cannot view the password ?

![Screenshot 2024-04-24 at 9 34 16 AM](https://github.com/keepassxreboot/keepassxc/assets/2174170/dae4a78c-9b77-4a5c-ad63-b8bc5b167a31)

**Reply by droidmonkey** [reply]
Have no idea, I'll debug the database you sent me. This is not a macos silicon issue, it is originating from the datagrip software.
